File: 1620-check-if-array-pairs-are-divisible-by-k/check-if-array-pairs-are-divisible-by-k.js
var canArrange = function(arr, k) {
    let lookup = {}

    for (let num of arr) {
        rem = ((num % k) + k) % k
        if (!(rem in lookup)) lookup[rem] = 0
        lookup[rem] += 1
    }    

    if (0 in lookup) {
        if (lookup[0] % 2 !== 0) return false
    }

    for (let i = 1; i < k; i++) {
        if (lookup[(i + k) % k] != lookup[(k - i + k) % k]) return false
    }

    return true
};

File: 1620-check-if-array-pairs-are-divisible-by-k/check-if-array-pairs-are-divisible-by-k.py
class Solution:
    def canArrange(self, arr: List[int], k: int) -> bool:
        lookup = {}

        for num in arr:
            rem = ((num % k) + k) % k

            if (rem not in lookup):
                lookup[rem] = 0
            lookup[rem] += 1

        if (0 in lookup):
            if (lookup[0] % 2 != 0):
                return False
        
        for i in range(1, k):
            if ((i + k) % k not in lookup):
                continue
            if (k-i + k) % k not in lookup:
                return False
            if ((lookup[(i + k) % k] != lookup[(k - i + k) % k])):
                return False

        return True
        